README.md

```diff
@@ -30,165 +32,555 @@ The two guiding principles are:
+```javascript
+import React from 'react';
+export default () => <p>This has been loaded lazily</p>;
+```
 
-If you prefer, you can also explicitly list the presets that you want to use under the `presets` key in your [application configuration](#configuration). This will disable automatic discovery of hops presets.
+**`src/app.js`**
+
+```javascript
+import { render, Import } from 'hops';
+import React from 'react';
+
+const MyLazyComponent = Import('./my-component');
+
+export default render(
+  <div>
+    <h1>Hello World</h1>
+    <MyLazyComponent />
+  </div>
+);
+```
+
+### CLI / npm scripts
+
+The `hops` package acts as a CLI to be used inside [npm scripts](https://docs.npmjs.com/cli/run-script) or [npx](https://medium.com/@maybekatz/introducing-npx-an-npm-package-runner-55f7d4bd282b):
+
+**`package.json`**
+
+```json
+{
+  "scripts": {
+    "start": "hops start",
+    "build": "hops build",
+    "serve": "hops serve"
+  }
+}
+```
+
+#### `hops --help`
+
+The Hops CLI provides short help texts for all its commands. If, for example, you want to see what options are available for the `build` command you can type: `npx hops build --help`.
+
+#### `npm start`
+
+This command will either start the development server or execute a production build and start the production server, depending on the environment variable `NODE_ENV` and/or the `--production` flag.
+
+_The production flag (`--production` or `-p`) is a shortcut for setting the environment variable `NODE_ENV=production`, so these two are interchangable._
+
+- `npm start` starts the development server with hot module reloading, etc.
+- `npm start -- --production` executes a production build and then starts the production server.
+- `npm start -- --production --static` exports static HTML pages for all configured [locations](#default-settings) and then starts the production server (_this only works in tandem with production mode_).
+
+#### `npm run build`
+
+This command will execute a single build of all your assets.
+
+- `npm run build` builds all assets in development mode.
+- `npm run build -- --production` builds all assets in production mode (which includes minification, etc).
+- `npm run build -- --static` export static HTML pages for all configured [locations](#default-settings) in development mode.
+- `npm run build -- --production --static` export static HTML pages in production mode (enables minification, etc).
+
+#### `npm run serve`
+
+Starts a production ready Express.js server.
+
+- `npm run serve` starts the server in development mode (without optimizations).
+- `npm run serve -- --production` starts the server in production mode (enables gzip and other optimizations).
+
+## Configuration
+
+### Settings
+
+**TODO:** write a small sentence explaining what settings actually are.
+
+You can provide settings to a Hops application via either:
+
+- a `"hops"` key in your `package.json`
```

Comment:
Maybe — to keep the focus of the reader on the actual settings — we should only tell them here, that settings are done via the `"hops"`-key in the `package.json`. Then after they read through the table and understood what this is all about, we can go deeper into the topic and mention the use of cosmiconfig and different settings approaches.

```markdown
You can provide settings to a Hops application via a `"hops"` key in your `package.json`.

… [table]

Under the hood Hops uses [`cosmiconfig`](https://github.com/davidtheclark/cosmiconfig) to gather
settings. So you're not limited to the `"hops"` key in your `package.json`, but can alternatively
use an external settings file in the root directory of your project. The filename then has to be
`hops.config.js` or `.hopsrc{.json,.yaml,.js}`.
```